tests/conditional_compilation/requirements.txt

```diff
@@ -0,0 +1 @@
+pytest-dependency==0.5.1
```

Comment:
Perhaps it would be better to use https://github.com/pytest-dev/pytest-order instead of `pytest-dependency` since the first one is a part of `pytest-dev` (and looks like more supported). Anyway, it also provides functionality from `pytest-dependency`

@asomsiko what do you think?

Reply:
@vurusovs, @asomsiko, review again please

Reply:
Can be resolved in order to unblock merge